src/cuda/poisson_fft.f90

```diff
@@ -214,32 +142,23 @@ subroutine fft_postprocess_cuda(self)
 
     class(cuda_poisson_fft_t) :: self
 
-    complex(dp), device, dimension(:, :, :), pointer :: c_dev, c_x_ptr
+    complex(dp), device, dimension(:, :, :), pointer :: c_dev
     type(dim3) :: blocks, threads
+    integer :: tsize
 
-    ! Get some complex array pointers with right shape
-    c_x_ptr(1:self%nx, 1:SZ, 1:(self%ny*(self%nz/2 + 1))/SZ) => self%c_x_dev
-    c_dev(1:SZ, 1:self%nx, 1:(self%ny*(self%nz/2 + 1))/SZ) => self%c_y_dev
+    ! tsize is different than SZ, because here we work on a 3D Cartesian
+    ! data structure, and free to specify any suitable thread/block size.
+    tsize = 16
```

Comment:
If this might be tuned later, can it be stored more publicly? E.g. as a parameter in `common.f90`. I'm worried this parameter will get forgotten later.

Reply:
At the moment the only use case for this variable is here in the postprocessing, and for this specific kernel. GPU kernels are generally written considering the block and thread sizes, so I think its a good idea to define it here and use right after. For example, if we need to process a 3D Cartesian array its likely that we'll require a different thread size. The variable here works well because the kernel we have requires 3 separate 1D arrays with global size to carry out the operation, probably we won't need a similar operation anywhere else. If we observe use of this elsewhere then maybe we can move somewhere else later on.